### PR TITLE
Convert to using ansible_runner module

### PR DESCRIPTION
**Pull request by dustinblack**
## Changes introduced with this PR

The ansible python module is considered internal and not stable. This converts the metadata collection to using the supported ansible_runner module. As a side effect, this drastically simplifies (and seems to speed up) the plugin.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ansible-core==2.13.5; python_version >= "3.8"
 ansible==6.5.0; python_version >= "3.8"
+ansible_runner
```
Comment (portante) [inline requirements.txt:3]: Isn't this managed by `poetry` now?  As such, we want to remove the `requirements.txt` and just update the `pyproject.toml`.  Let me know if you want some help with that.

**Review by portante — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (dustinblack) [reply requirements.txt:3]: This module has not been updated to poetry. I would rather make that change in a separate PR.

**Review by jaredoconnell — APPROVED** [review]
Looks good to me. Looks like a more sustainable way to do it.

File: requirements.txt (hunk 1 above)
Comment (jaredoconnell) [reply requirements.txt:3]: I also added the dependency to the pyproject.toml. Another PR can switch the Dockerfile to poetry.

Comment (portante) [reply requirements.txt:3]: Why are we not regenerating the requirements.txt file using `poetry`?

**Review by portante — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (portante) [reply requirements.txt:3]: Basically, you need to do the following:

```
poetry remove ansible
poetry add ansible_runner
poetry update
poetry export > requirements.txt
git add pyproject.toml requirements.txt
git commit -m 'Replace `ansible` with `ansible_runner`'
```

Comment (portante) [reply requirements.txt:3]: FWIW, once `poetry` is properly updated, and the `requirements.txt` file generated, the python module that shows up in the updated `requirements.txt` file is `ansible-runner`, not `ansible_runner`.

Comment (jaredoconnell) [reply requirements.txt:3]: > Why are we not regenerating the requirements.txt file using `poetry`?

In a following PR we'll remove requirements.txt in favor of exclusively poetry, so this is just temporary. That PR will be the one that switches it to the new format for the Dockerfile, which will build the requirements.txt in a build step.

Comment (portante) [reply requirements.txt:3]: The requirements.txt file is wrong for `ansible_runner`.  We can't separate these changes.  Why can't we just do the above steps and commit this PR?

Comment (dustinblack) [reply requirements.txt:3]: They are functionally distinct changes that I would rather not munge into a single PR. The `requrements.txt` file as provided passes both manual and automated tests. I don't see what stops these from being distinct changes.

Comment (dustinblack) [reply requirements.txt:3]: > ```
> poetry remove ansible
> ```

The ansible plugin is still used for one function.

Comment (portante) [reply requirements.txt:3]: Why does the fact that it passes manual and automated tests say it is working?  If the file contents are incorrect, isn't it possible that the incorrect contents are making it pass?

I'll post a PR with the rebuilt changes so we can evaluate.

Comment (dustinblack) [reply requirements.txt:3]: > FWIW, once `poetry` is properly updated, and the `requirements.txt` file generated, the python module that shows up in the updated `requirements.txt` file is `ansible-runner`, not `ansible_runner`.

I tested this, and both `ansible_runner` and `ansible-runner` in the `requirements.txt` file work with the same result.

Comment (dustinblack) [reply requirements.txt:3]: > Why does the fact that it passes manual and automated tests say it is working? If the file contents are incorrect, isn't it possible that the incorrect contents are making it pass?

I don't understand your question. That is exactly what tests are for.

Comment (portante) [reply requirements.txt:3]: See PR https://github.com/arcalot/arcaflow-plugin-metadata/pull/11.

Comment (portante) [reply requirements.txt]: And PR #12.

**Review by portante — CHANGES_REQUESTED** [review]
See PR #12.

**Review by portante — APPROVED** [review]

**Pull request merged**